app/src/main/java/org/wikipedia/page/leadimages/LeadImagesHandler.java

```diff
@@ -168,11 +194,65 @@ private void loadLeadImage(@Nullable String url) {
         if (!isMainPage() && !TextUtils.isEmpty(url) && isLeadImageEnabled()) {
             pageHeaderView.show(isLeadImageEnabled());
             pageHeaderView.loadImage(url);
+            updateCallToAction();
         } else {
             pageHeaderView.loadImage(null);
         }
     }
 
+    private void updateCallToAction() {
+        if ((!Prefs.isSuggestedEditsAddCaptionsUnlocked() && !Prefs.isSuggestedEditsTranslateCaptionsUnlocked()) || !getLeadImageUrl().contains(COMMONS_IMAGE_URL) || getPage() == null) {
+            pageHeaderView.setUpCallToAction(null);
+            return;
+        }
+        WikipediaApp app = WikipediaApp.getInstance();
+        String imageName = getPage().getPageProperties().getLeadImageName();
+        String filename = "File:" + imageName;
+
+        disposables.add(ServiceFactory.getRest(getTitle().getWikiSite()).getMedia(getTitle().getConvertedText())
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(gallery -> {
+                    List<GalleryItem> list = gallery.getItems("image");
+                    for (GalleryItem item : list) {
+                        if (getPage() != null && item.getFilePage().contains(getPage().getPageProperties().getLeadImageName())) {
+                            if (TextUtils.isEmpty(item.getStructuredCaptions().get(getTitle().getWikiSite().languageCode()))) {
+                                pageHeaderView.setUpCallToAction(app.getResources().getString(R.string.suggested_edits_article_cta_add_image_caption));
+                                sourceSummary = new SuggestedEditsSummary(getTitle().getPrefixedText(), app.getAppOrSystemLanguageCode(), getTitle(),
+                                        getTitle().getDisplayText(), getTitle().getDisplayText(), StringUtil.fromHtml(item.getDescription().getHtml()).toString(), item.getThumbnailUrl(), item.getPreferredSizedImageUrl(),
+                                        null, null, null, null);
+
+                                return;
+                            }
+                        }
+                    }
+                    if (app.language().getAppLanguageCodes().size() > 1 && Prefs.isSuggestedEditsTranslateCaptionsUnlocked()) {
+                        imageCaptionDisposable = MediaHelper.INSTANCE.getImageCaptions(filename)
```

Comment:
Can we use `map` or `flapMap` for the logic? I don't think to have another `disposable` inside a `disposable` is a good idea.